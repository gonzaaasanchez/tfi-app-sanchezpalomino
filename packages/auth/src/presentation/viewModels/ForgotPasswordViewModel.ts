import { UIState, useInjection } from '@app/common'
import { useState } from 'react'
import { $ } from '../../domain/di/Types'
import { ForgotPasswordUseCase } from '../../domain/usecases/ForgotPasswordUseCase'

/**
 * Represents the view model for a forgot password page.
 *
 * @notExported
 */
type ForgotPasswordViewModel = {
  /**
   * The current state of the view model.
   */
  state: ForgotPasswordState

  /**
   * Sends a forgot password request for the given email and returns a promise that resolves to `true` if the request is successful.
   * @returns {Promise<boolean>} A promise that resolves to `true` if the request is successful.
   */
  forgotPassword: () => Promise<boolean>

  /**
   * Sets the email field of the view model's state to a new email address.
   * @param {string} email - The new email address.
   */
  setEmail: (email: string) => void
}

/**
 * Represents the state of a forgot password view model.
 *
 * @property {string} email - The user's email address.
 * @property {string} password - The user's new password.
 * @extends {UIState} - Additional UI state properties.
 * @notExported
 */
type ForgotPasswordState = {
  email: string
  password: string
} & UIState

const initialState: ForgotPasswordState = {
  loading: false,
  error: null,
  email: '',
  password: '',
}

/**
 * Creates a ForgotPasswordViewModel that handles the forgot password feature.
 *
 * The ForgotPasswordViewModel provides the following properties:
 * - state: The current state of the view model, which includes the loading
 *   state, error state, and the email.
 * - forgotPassword: A function that sends the forgot password request for
 *   the given email.
 * - setEmail: A function that sets the email field of the view model's state.
 *
 * @return {ForgotPasswordViewModel} The ForgotPasswordViewModel instance.
 */
const useForgotPasswordViewModel = (): ForgotPasswordViewModel => {
  const useCase: ForgotPasswordUseCase = useInjection($.ForgotPasswordUseCase)
  const [state, setState] = useState<ForgotPasswordState>(initialState)

  /**
   * Sends the forgot password request for the given email.
   *
   * Sets the loading state to true, then sets the error state to null. If the
   * AuthRepository call succeeds, it sets the loading state to false and
   * returns true. If the AuthRepository call fails, it sets the loading state to
   * false and sets the error state to the error message.
   *
   * @return {Promise<boolean>} A promise that resolves to true if the request
   *   was sent successfully, or rejects with an error if the request failed.
   */
  const forgotPassword: () => Promise<boolean> = async () => {
    setState((previous) => ({ ...previous, loading: true, error: null }))
    try {
      await useCase.execute(state.email)
      setState((previous) => ({ ...previous, loading: false }))
      return true
    } catch (error) {
      setState((previous) => ({
        ...previous,
        loading: false,
        error: error.message,
      }))
    }
  }

  /**
   * Sets the email field of the view model's state.
   *
   * @param email The new email address.
   */
  const setEmail: (email: string) => void = async (email) => {
    setState((previous) => ({ ...previous, email: email }))
  }

  return { state, forgotPassword, setEmail }
}

export { useForgotPasswordViewModel }
