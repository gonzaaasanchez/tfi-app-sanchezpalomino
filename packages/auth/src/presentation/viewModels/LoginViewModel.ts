import { setAuthToken, UIState, useInjection } from '@app/common'
import { useState } from 'react'
import { useDispatch } from 'react-redux'
import { $ } from '../../domain/di/Types'
import { LoginUseCase } from '../../domain/usecases/LoginUseCase'

/**
 * Represents the view model for the login feature.
 *
 * @notExported
 */
type LoginViewModel = {
  /**
   * The current state of the login view model.
   */
  state: LoginState
  /**
   * Triggers the login process.
   */
  login: () => void
  /**
   * Sets the email address in the view model's state.
   *
   * @param email The email address to set.
   */
  setEmail: (email: string) => void
  /**
   * Sets the password in the view model's state.
   *
   * @param password The password to set.
   */
  setPassword: (password: string) => void
}

/**
 * Represents the state of a login form.
 *
 * @property {string} email - The user's email address.
 * @property {string} password - The user's password.
 * @extends UIState
 * @notExported
 */
type LoginState = {
  email: string
  password: string
} & UIState

const initialState: LoginState = {
  loading: false,
  error: null,
  email: '',
  password: '',
}

/**
 * Creates a LoginViewModel that handles user login functionality.
 *
 * The LoginViewModel provides the following properties:
 * - state: The current state of the view model, which includes the loading
 *   state, error state, and the user's email and password.
 * - login: A function that logs in the user, dispatching the appropriate
 *   actions to update the Redux store and set the authentication token.
 * - setEmail: A function that sets the email field of the view model's state.
 * - setPassword: A function that sets the password field of the view model's
 *   state.
 *
 * @return {LoginViewModel} The LoginViewModel instance.
 */
const useLoginViewModel = (): LoginViewModel => {
  const dispatch = useDispatch()
  const useCase: LoginUseCase = useInjection($.LoginUseCase)
  const [state, setState] = useState<LoginState>(initialState)

  /**
   * Logs in the user using the email and password from the state. If either
   * the email or password is blank, an alert is displayed with an error
   * message. Otherwise, the login function from the AuthRepository is called
   * to authenticate the user.
   */
  const login: () => void = async () => {
    setState((previous) => ({ ...previous, loading: true, error: null }))
    try {
      const user = await useCase.execute(state.email, state.password)
      setState((previous) => ({ ...previous, loading: false, error: null }))
      dispatch(setAuthToken({ token: user.token }))
    } catch (error) {
      setState((previous) => ({
        ...previous,
        loading: false,
        error: error.message,
      }))
    }
  }

  /**
   * Sets the email field of the view model's state.
   *
   * @param email The new email address.
   */
  const setEmail: (email: string) => void = async (email) => {
    setState((previous) => ({ ...previous, email: email }))
  }

  /**
   * Sets the password field of the view model's state.
   *
   * @param password The new password.
   */
  const setPassword: (password: string) => void = async (password) => {
    setState((previous) => ({ ...previous, password: password }))
  }

  return { state, login, setEmail, setPassword }
}

export { useLoginViewModel }
