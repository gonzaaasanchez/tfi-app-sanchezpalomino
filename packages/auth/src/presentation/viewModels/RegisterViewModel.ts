import { useDispatch } from 'react-redux'
import { UIState, useInjection } from '@app/common'
import { setAuthToken } from '@app/common'
import { useState } from 'react'
import { $ } from '../../domain/di/Types'
import { RegisterUseCase } from '../../domain/usecases/RegisterUseCase'

/**
 * Represents the view model for the user registration process.
 *
 * @notExported
 */
type RegisterViewModel = {
  /**
   * The current state of the registration process.
   */
  state: RegisterState
  /**
   * Creates a new user by dispatching actions to update the Redux store and set the authentication token.
   */
  createUser: () => Promise<boolean>
  /**
   * Sets the email field of the view model's state.
   */
  setEmail: (email: string) => void
  /**
   * Sets the password field of the view model's state.
   */
  setPassword: (password: string) => void
  /**
   * Sets the confirm password field of the view model's state.
   */
  setConfirmPassword: (password: string) => void
  /**
   * Sets the name field of the view model's state.
   */
  setName: (name: string) => void
}

/**
 * Represents the state of the registration process.
 *
 * @property {string} email - The user's email address.
 * @property {string} password - The user's password.
 * @property {string} confirmPassword - The user's confirmed password.
 * @property {string} name - The user's name.
 * @extends UIState
 * @notExported
 */
type RegisterState = {
  email: string
  password: string
  confirmPassword: string
  name: string
} & UIState

const initialState: RegisterState = {
  loading: false,
  error: null,
  email: '',
  password: '',
  confirmPassword: '',
  name: '',
}

/**
 * Creates a RegisterViewModel that handles the user registration process.
 *
 * The RegisterViewModel provides the following properties:
 * - state: The current state of the view model, which includes the loading
 *   state, error state, email, password, confirm password, and name.
 * - createUser: A function that registers a new user with the provided email,
 *   name, password, and confirm password. If the registration is successful,
 *   it dispatches actions to update the Redux store and set the authentication
 *   token.
 * - setEmail: A function that sets the email field of the view model's state.
 * - setPassword: A function that sets the password field of the view model's state.
 * - setConfirmPassword: A function that sets the confirm password field of the view
 *   model's state.
 * - setName: A function that sets the name field of the view model's state.
 *
 * @return {RegisterViewModel} The RegisterViewModel instance.
 */
const useRegisterViewModel = (): RegisterViewModel => {
  const useCase: RegisterUseCase = useInjection($.RegisterUseCase)
  const dispatch = useDispatch()
  const [state, setState] = useState<RegisterState>(initialState)

  /**
   * Registers a new user and dispatches actions to update the Redux store and set the
   * authentication token.
   *
   * @returns {Promise<boolean>} A promise that resolves to true if the user was created
   *   successfully, or rejects with an error if the user creation failed.
   */
  const createUser: () => Promise<boolean> = async () => {
    setState((previous) => ({ ...previous, loading: true, error: null }))
    try {
      const user = await useCase.execute(
        state.email,
        state.name,
        state.password,
        state.confirmPassword
      )
      setState((previous) => ({ ...previous, loading: false, error: null }))
      dispatch(setAuthToken({ token: user.token }))
      return true
    } catch (error) {
      setState((previous) => ({
        ...previous,
        loading: false,
        error: error.message,
      }))
    }
  }

  /**
   * Sets the email field of the view model's state.
   *
   * @param email The new email address.
   */
  const setEmail: (email: string) => void = async (email) => {
    setState((previous) => ({ ...previous, email: email }))
  }

  /**
   * Sets the password field of the view model's state.
   *
   * @param password The new password.
   */
  const setPassword: (password: string) => void = async (password) => {
    setState((previous) => ({ ...previous, password: password }))
  }

  /**
   * Sets the confirm password field of the view model's state.
   *
   * @param password The new confirm password.
   */
  const setConfirmPassword: (password: string) => void = async (password) => {
    setState((previous) => ({ ...previous, confirmPassword: password }))
  }

  /**
   * Sets the name field of the view model's state.
   *
   * @param name The new name.
   */
  const setName: (name: string) => void = async (name) => {
    setState((previous) => ({ ...previous, name: name }))
  }

  return {
    state,
    createUser,
    setEmail,
    setPassword,
    setConfirmPassword,
    setName,
  }
}

export { useRegisterViewModel }
