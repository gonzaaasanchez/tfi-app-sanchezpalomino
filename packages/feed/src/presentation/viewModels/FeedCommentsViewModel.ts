import {
  UIState,
  useInjection,
  usePagination,
  PaginationModel,
  LoadFunction,
  useI18n,
  ShowToast,
  Types,
} from '@packages/common'
import { useState, useEffect, useCallback } from 'react'
import { GetFeedCommentsUseCase } from '../../domain/usecases/GetFeedCommentsUseCase'
import { CreateCommentUseCase } from '../../domain/usecases/CreateCommentUseCase'
import { $ } from '../../domain/di/Types'
import { CommentModel } from '@packages/common'

type FeedCommentsState = {
  postId: string
  commentText: string
  pagination: {
    items: CommentModel[]
    pagination: PaginationModel
    loading: boolean
    loadingMore: boolean
  }
} & UIState

type FeedCommentsViewModel = {
  state: FeedCommentsState
  baseUrl: string
  loadComments: ({ reset }: { reset: boolean }) => Promise<void>
  setCommentText: (text: string) => void
  createComment: () => Promise<void>
}

const useFeedCommentsViewModel = (postId: string): FeedCommentsViewModel => {
  const [state, setState] = useState<FeedCommentsState>({
    loading: false,
    error: null,
    postId,
    commentText: '',
    pagination: {
      items: [],
      pagination: {
        page: 1,
        limit: 10,
        total: 0,
        totalPages: 0,
      },
      loading: false,
      loadingMore: false,
    },
  })

  const getFeedCommentsUseCase: GetFeedCommentsUseCase = useInjection(
    $.GetFeedCommentsUseCase
  )
  const createCommentUseCase: CreateCommentUseCase = useInjection(
    $.CreateCommentUseCase
  )
  const { t } = useI18n()
  const baseUrl = useInjection(Types.BaseURL) as string

  // Create load function for pagination hook
  const loadCommentsFunction: LoadFunction<CommentModel> = useCallback(
    async (page: number, limit: number) => {
      const response = await getFeedCommentsUseCase.execute(postId, page, limit)

      return {
        items: response.items || [],
        pagination: response.pagination,
      }
    },
    [getFeedCommentsUseCase, postId]
  )

  const pagination = usePagination(loadCommentsFunction)

  const loadComments = async ({ reset }: { reset: boolean }): Promise<void> => {
    try {
      setState((previous) => ({
        ...previous,
        loading: true,
        error: null,
      }))

      await pagination.loadItems(reset)
    } catch (error) {
      setState((previous) => ({
        ...previous,
      }))
      ShowToast({
        config: 'error',
        title: t('general.ups'),
        subtitle:
          error instanceof Error
            ? error.message
            : 'Error al cargar los comentarios',
      })
    } finally {
      setState((previous) => ({
        ...previous,
        loading: false,
      }))
    }
  }

  useEffect(() => {
    loadComments({ reset: true })
  }, [postId])

  const setCommentText = (text: string): void => {
    setState((previous) => ({
      ...previous,
      commentText: text,
    }))
  }

  const createComment = async (): Promise<void> => {
    if (!state.commentText.trim()) return

    try {
      setState((previous) => ({
        ...previous,
        loading: true,
        error: null,
      }))

      await createCommentUseCase.execute(postId, state.commentText.trim())

      setState((previous) => ({
        ...previous,
        commentText: '',
      }))

      await loadComments({ reset: true })
    } catch (error) {
      setState((previous) => ({
        ...previous,
      }))
      ShowToast({
        config: 'error',
        title: t('general.ups'),
        subtitle:
          error instanceof Error
            ? error.message
            : 'Error al crear el comentario',
      })
    } finally {
      setState((previous) => ({
        ...previous,
        loading: false,
      }))
    }
  }

  return {
    state: {
      ...state,
      pagination: {
        items: pagination.state.items,
        pagination: pagination.state.pagination,
        loading: pagination.state.loading,
        loadingMore: pagination.state.loadingMore,
      },
    },
    baseUrl,
    loadComments,
    createComment,
    setCommentText,
  }
}

export { useFeedCommentsViewModel }
