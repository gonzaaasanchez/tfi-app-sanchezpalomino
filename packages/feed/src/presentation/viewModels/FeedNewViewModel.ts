import {
  UIState,
  useI18n,
  ShowToast,
  Color,
  LabelStyle,
  useInjection,
  useImagePicker,
} from '@packages/common'
import { useState } from 'react'
import { useDispatch } from 'react-redux'
import { $ } from '../../domain/di/Types'
import { CreatePostUseCase } from '../../domain/usecases/CreatePostUseCase'
import { markPostCreated } from '../../domain/store/FeedSlice'
import { useNavigation } from '@react-navigation/native'

type FeedNewState = {
  title: string | null
  description: string | null
  avatarFile: string | null
} & UIState

type FeedNewViewModel = {
  state: FeedNewState
  setTitle: (title: string) => void
  setDescription: (description: string) => void
  selectImageFrom: (source: 'camera' | 'gallery') => Promise<void>
  validateForm: (onValidated: () => void) => void
  savePost: () => Promise<void>
}

const initialState: FeedNewState = {
  title: null,
  description: null,
  avatarFile: null,
  loading: false,
  error: null,
}

const useFeedNewViewModel = (): FeedNewViewModel => {
  const [state, setState] = useState<FeedNewState>(initialState)
  const { t } = useI18n()
  const dispatch = useDispatch()
  const navigation = useNavigation()
  const createPostUseCase: CreatePostUseCase = useInjection($.CreatePostUseCase)

  const {
    selectImageFromGallery: pickFromGallery,
    selectImageFromCamera: pickFromCamera,
  } = useImagePicker()

  const selectImageFrom = async (
    source: 'camera' | 'gallery'
  ): Promise<void> => {
    const { uri, error } =
      source === 'camera' ? await pickFromCamera() : await pickFromGallery()

    if (error) {
      ShowToast({
        config: 'error',
        title: t('general.ups'),
        subtitle: error,
      })
      return
    }

    if (uri) {
      setState((prev) => ({
        ...prev,
        avatarFile: uri,
      }))
    }
  }

  const setTitle = (title: string): void => {
    setState((previous) => ({
      ...previous,
      title,
    }))
  }

  const setDescription = (description: string): void => {
    setState((previous) => ({
      ...previous,
      description,
    }))
  }

  const validateForm = (onValidated: () => void): void => {
    const errors = getValidationErrors()

    if (errors.length === 0) {
      onValidated()
    } else {
      const errorMessage = errors.map((error) => `â€¢ ${error}`).join('\n')
      ShowToast({
        config: 'error',
        title: t('general.ups'),
        subtitle: errorMessage,
        duration: 5000,
        subtitleStyle: {
          ...LabelStyle.body({
            fontWeight: 400,
            color: Color.black[600],
          }),
        },
      })
    }
  }

  const getValidationErrors = (): string[] => {
    const errors: string[] = []
    const { title, description, avatarFile } = state

    if (!title?.trim()) {
      errors.push(t('feedNewScreen.validation.titleRequired'))
    }

    if (!description?.trim()) {
      errors.push(t('feedNewScreen.validation.descriptionRequired'))
    }

    if (!avatarFile) {
      errors.push(t('feedNewScreen.validation.imageRequired'))
    }

    return errors
  }

  const savePost = async (): Promise<void> => {
    setState((previous) => ({
      ...previous,
      loading: true,
      error: null,
    }))

    try {
      await createPostUseCase.execute({
        title: state.title,
        description: state.description,
        image: state.avatarFile,
      })

      setState((previous) => ({
        ...previous,
        loading: false,
      }))

      ShowToast({
        config: 'success',
        title: t('feedNewScreen.success.title'),
        subtitle: t('feedNewScreen.success.subtitle'),
      })

      navigation.goBack()
      dispatch(markPostCreated())
    } catch (error) {
      setState((previous) => ({
        ...previous,
        loading: false,
      }))

      ShowToast({
        config: 'error',
        title: t('general.ups'),
        subtitle:
          error instanceof Error ? error.message : 'Error creating post',
      })
    }
  }

  return {
    state,
    setTitle,
    setDescription,
    selectImageFrom,
    validateForm,
    savePost,
  }
}

export { useFeedNewViewModel }
