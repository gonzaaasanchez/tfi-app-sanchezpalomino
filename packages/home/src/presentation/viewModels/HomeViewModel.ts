import { UIState } from '@app/common'
import { TestModel } from '../../data/models/TestModel'
import { useState } from 'react'

type HomeViewModel = {
  state: HomeState
}
type HomeState = {
  data: TestModel
} & UIState

const initialState: HomeState = {
  loading: false,
  error: null,
  data: { test: '' },
}

const useHomeViewModel = (): HomeViewModel => {
  const [state] = useState<HomeState>(initialState)

  return { state: { ...state } }
}

export { useHomeViewModel }
