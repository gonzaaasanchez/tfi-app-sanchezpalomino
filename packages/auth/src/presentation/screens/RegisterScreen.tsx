import React, { useEffect, useLayoutEffect } from 'react'
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  KeyboardAvoidingView,
  Platform,
  Alert,
} from 'react-native'
import { ParamList } from '../AuthStack'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import { useRegisterViewModel } from '../viewModels/RegisterViewModel'
import { useI18n } from '@app/common'

type Props = NativeStackScreenProps<ParamList, 'registerScreen'>

/**
 * RegisterScreen component.
 *
 * This component renders a registration screen that allows users to enter their
 * name, email, password and confirm password, and then log in or navigate to login
 * screen.
 *
 * @param {Props} props - Component props.
 * @returns {JSX.Element} The rendered RegisterScreen component.
 */
const RegisterScreen = ({ route }: Props): JSX.Element => {
  const {
    state,
    createUser,
    setEmail,
    setPassword,
    setConfirmPassword,
    setName,
  } = useRegisterViewModel()

  const { email } = route.params
  const { t } = useI18n()

  /**
   * Handles the register button press.
   *
   * Calls the createUser function from the useRegisterViewModel hook. If the
   * createUser function succeeds, it displays a success alert.
   *
   * @returns {Promise<void>}
   */
  const handleRegister: () => Promise<void> = async () => {
    if (await createUser()) {
      Alert.alert(
        t('registerScreen.success.title'),
        t('registerScreen.success.message')
      )
    }
  }

  useLayoutEffect(() => {
    setEmail(email)
  }, [])

  useEffect(() => {
    if (state.error === 'register-missing-fields') {
      Alert.alert(
        t('registerScreen.error.title'),
        t('registerScreen.error.missingFields')
      )
    } else if (state.error === 'register-password-not-match') {
      Alert.alert(
        t('registerScreen.error.title'),
        t('registerScreen.error.passwordNotMatch')
      )
    } else if (state.error !== null) {
      Alert.alert(t('registerScreen.error.title'), state.error)
    }
  }, [state.error])

  if (state.loading === true) {
    return (
      <View style={styles.loading}>
        <Text>{t('registerScreen.loading')}</Text>
      </View>
    )
  }

  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
    >
      <View style={styles.inner} accessible={false}>
        <Text style={styles.title} testID="registerScreen.title">
          {t('registerScreen.title')}
        </Text>

        <TextInput
          style={styles.input}
          placeholder={t('registerScreen.name')}
          placeholderTextColor="#aaa"
          value={state.name}
          onChangeText={setName}
          autoCapitalize="words"
          autoCorrect={false}
          testID="registerScreen.name"
        />

        <TextInput
          style={styles.input}
          placeholder={t('registerScreen.email')}
          placeholderTextColor="#aaa"
          keyboardType="email-address"
          value={state.email}
          onChangeText={setEmail}
          autoCapitalize="none"
          autoCorrect={false}
          textContentType="emailAddress"
          testID="registerScreen.email"
        />

        <TextInput
          style={styles.input}
          placeholder={t('registerScreen.password')}
          placeholderTextColor="#aaa"
          secureTextEntry
          value={state.password}
          onChangeText={setPassword}
          autoCapitalize="none"
          autoCorrect={false}
          textContentType={'oneTimeCode'}
          testID="registerScreen.password"
        />

        <TextInput
          style={styles.input}
          placeholder={t('registerScreen.confirmPassword')}
          placeholderTextColor="#aaa"
          secureTextEntry
          value={state.confirmPassword}
          onChangeText={setConfirmPassword}
          autoCapitalize="none"
          autoCorrect={false}
          textContentType={'oneTimeCode'}
          testID="registerScreen.confirmPassword"
        />

        <TouchableOpacity
          style={styles.button}
          onPress={handleRegister}
          testID="registerScreen.register"
        >
          <Text style={styles.buttonText}>{t('registerScreen.register')}</Text>
        </TouchableOpacity>
      </View>
    </KeyboardAvoidingView>
  )
}

const styles = StyleSheet.create({
  loading: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#f8f9fa',
  },
  inner: {
    padding: 24,
    flex: 1,
    justifyContent: 'center',
  },
  title: {
    fontSize: 32,
    fontWeight: '600',
    marginBottom: 24,
    color: '#333',
    textAlign: 'center',
  },
  input: {
    height: 48,
    borderColor: '#ddd',
    borderWidth: 1,
    borderRadius: 8,
    paddingHorizontal: 16,
    marginBottom: 16,
    backgroundColor: '#fff',
  },
  button: {
    height: 48,
    backgroundColor: '#007bff',
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 16,
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: '600',
  },
})

export default RegisterScreen
