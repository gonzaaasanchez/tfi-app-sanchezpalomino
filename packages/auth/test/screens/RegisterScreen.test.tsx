import React from 'react'
import { render } from '@testing-library/react-native'
import RegisterScreen from '../../src/presentation/screens/RegisterScreen'
import { useRegisterViewModel } from '../../src/presentation/viewModels/RegisterViewModel'
import { NavigationContainer } from '@react-navigation/native'

jest.mock('../../src/presentation/viewModels/RegisterViewModel', () => ({
  useRegisterViewModel: jest.fn(),
}))

describe('RegisterScreen', () => {
  it('should match snapshot when loading', () => {
    ;(useRegisterViewModel as jest.Mock).mockReturnValue({
      state: {
        loading: true,
        name: '',
        email: '',
        password: '',
        confirmPassword: '',
        error: null,
      },
      createUser: jest.fn(),
      setEmail: jest.fn(),
      setPassword: jest.fn(),
      setConfirmPassword: jest.fn(),
      setName: jest.fn(),
    })

    const { toJSON } = render(
      <NavigationContainer>
        <RegisterScreen
          navigation={undefined}
          route={{
            key: 'registerScreen',
            name: 'registerScreen',
            params: { email: '' },
          }}
        />
      </NavigationContainer>
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should match snapshot with default state', () => {
    ;(useRegisterViewModel as jest.Mock).mockReturnValue({
      state: {
        loading: false,
        name: '',
        email: '',
        password: '',
        confirmPassword: '',
        error: null,
      },
      createUser: jest.fn(),
      setEmail: jest.fn(),
      setPassword: jest.fn(),
      setConfirmPassword: jest.fn(),
      setName: jest.fn(),
    })

    const { toJSON } = render(
      <NavigationContainer>
        <RegisterScreen
          navigation={undefined}
          route={{
            key: 'registerScreen',
            name: 'registerScreen',
            params: { email: '' },
          }}
        />
      </NavigationContainer>
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
