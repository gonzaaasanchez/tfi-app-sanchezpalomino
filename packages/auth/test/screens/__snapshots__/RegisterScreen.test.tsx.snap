// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen should match snapshot when loading 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    registerScreen.loading
  </Text>
</View>
`;

exports[`RegisterScreen should match snapshot with default state 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "backgroundColor": "#f8f9fa",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessible={false}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "padding": 24,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 32,
          "fontWeight": "600",
          "marginBottom": 24,
          "textAlign": "center",
        }
      }
      testID="registerScreen.title"
    >
      registerScreen.title
    </Text>
    <TextInput
      autoCapitalize="words"
      autoCorrect={false}
      onChangeText={[MockFunction]}
      placeholder="registerScreen.name"
      placeholderTextColor="#aaa"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 48,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        }
      }
      testID="registerScreen.name"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="email-address"
      onChangeText={
        [MockFunction] {
          "calls": [
            [
              "",
            ],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      placeholder="registerScreen.email"
      placeholderTextColor="#aaa"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 48,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        }
      }
      testID="registerScreen.email"
      textContentType="emailAddress"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[MockFunction]}
      placeholder="registerScreen.password"
      placeholderTextColor="#aaa"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 48,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        }
      }
      testID="registerScreen.password"
      textContentType="oneTimeCode"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[MockFunction]}
      placeholder="registerScreen.confirmPassword"
      placeholderTextColor="#aaa"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 48,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        }
      }
      testID="registerScreen.confirmPassword"
      textContentType="oneTimeCode"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#007bff",
          "borderRadius": 8,
          "height": 48,
          "justifyContent": "center",
          "marginTop": 16,
          "opacity": 1,
        }
      }
      testID="registerScreen.register"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        registerScreen.register
      </Text>
    </View>
  </View>
</View>
`;
